File: daily-coding-challenge/2026/july/js/27_PronicNumber#351.js
/*

Pronic Number

Given a number, determine whether it is a pronic number.

A pronic number is the product of two consecutive integers. For example, 6 is pronic because 2 * 3 = 6.

Tests:
  isPronic(6) should return true.
  isPronic(15) should return false.
  isPronic(12) should return true.
  isPronic(132) should return true.
  isPronic(80) should return false.
  isPronic(0) should return true.

*/

function isPronic(n) {
  if (n < 0) return false;

  let root = Math.ceil(Math.sqrt(n)); // floor

  return root * (root - 1) === n; // + 1
}

console.log(isPronic(6)); // true.
console.log(isPronic(15)); // false.
console.log(isPronic(12)); // true.
console.log(isPronic(132)); // true.
console.log(isPronic(80)); // false.
console.log(isPronic(0)); // true.

File: daily-coding-challenge/2026/july/python/27_PronicNumber#351.py
"""

Pronic Number

Given a number, determine whether it is a pronic number.

A pronic number is the product of two consecutive integers. For example, 6 is pronic because 2 * 3 = 6.

Tests:
  is_pronic(6) should return True.
  is_pronic(15) should return False.
  is_pronic(12) should return True.
  is_pronic(132) should return True.
  is_pronic(80) should return False.
  is_pronic(0) should return True.

"""


def is_pronic(n):
    if n < 0:
        return False

    root = int(n**0.5)  # math.isqrt(n)

    return root * (root + 1) == n


print(is_pronic(6))  # True.
print(is_pronic(15))  # False.
print(is_pronic(12))  # True.
print(is_pronic(132))  # True.
print(is_pronic(80))  # False.
print(is_pronic(0))  # True.
